src/utils/mod.rs

```diff
@@ -1524,3 +1626,176 @@ pub enum PublicKey {
     Rsa(Vec<u8>),
     Ecc { x: Vec<u8>, y: Vec<u8> },
 }
+
+/// Struct holding a pcr value.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PcrValue {
+    value: Vec<u8>,
+}
+
+impl PcrValue {
+    /// Function for retrieving the value.
+    pub fn value(&self) -> &[u8] {
+        &self.value
+    }
+}
+
+impl TryFrom<TPM2B_DIGEST> for PcrValue {
+    type Error = Error;
+    fn try_from(tss_digest: TPM2B_DIGEST) -> Result<Self> {
+        let size = tss_digest.size as usize;
+        if size > 64 {
+            error!("Error: Invalid TPM2B_DIGEST size(> 64)");
+            return Err(Error::local_error(WrapperErrorKind::InvalidParam));
+        }
+        Ok(PcrValue {
+            value: tss_digest.buffer[..size].to_vec(),
+        })
+    }
+}
+
+/// Struct for holding PcrSlots and their
+/// corresponding values.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct PcrBank {
+    bank: BTreeMap<PcrSlot, PcrValue>,
```

Comment:
I am curious of the use of a `BTreeMap` 😃 Is it more efficient than a `HashMap` in this case?

Reply:
No the reason for the BTreeMap here is because it is ordered. And if I am not mistaken when I read the specification is that the returned values from pcr_read are also in order. So the lowest selected slot number for a specific bank will end up first in the returned digest.

When I did this I did not take performance into account. I was more thinking that for the banks I did not care about ordering (= HashMap) for the pcr slot values I did care about the order because it made sense there (=BTreeMap).
If we are concerned about performance here then the HashMap needs to be removed. Because it will probably be slower for a small amount of items.

Reply:
Thanks for the explanation, I am good with it!